SubterfugeCoreTest/NameGenerator.test.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SubterfugeCore.Core.Generation;
+using SubterfugeCore.Core.Timing;
+
+namespace SubterfugeCoreTest
+{
+    [TestClass]
+    public class NameGenerator_test
+    {
+        private NameGenerator _generator;
+
+        [TestInitialize]
+        public void Setup()
+        {
+            _generator = new NameGenerator(new SeededRandom(1234));
+        }
+
+        [TestMethod]
+        public void GeneratesRandomNames()
+        {
+            string name1 = _generator.GetRandomName();
+            string name2 = _generator.GetRandomName();
+            Assert.IsTrue(name1.Equals(name2) == false);
+        }
+
+        [TestMethod]
+        public void NoDuplicatedNames()
+        {
+            List<string> strings = new List<string>();
+            while (_generator.HasNames())
+            {
+                strings.Add(_generator.GetRandomName());
+            }
+            
+            // Ensure no duplicates.
+            Assert.AreEqual(strings.Count, strings.Distinct().Count());
+        }
+
+        [TestMethod]
+        public void IfNoNamesLeftNoDuplicatesGenerated()
+        {
+            List<string> strings = new List<string>();
+            while (_generator.HasNames())
+            {
+                strings.Add(_generator.GetRandomName());
+            }
+            
+            // Generate 100 more names.
+            for (int i = 0; i < 100; i++)
+            {
+                strings.Add(_generator.GetRandomName());
+            }
+
+            // Ensure no duplicates.
+            Assert.AreEqual(strings.Count, strings.Distinct().Count());
```

Comment:
Ah, I suppose this is a variant of the comment I made before about a flakey test. This test _can_ and _will_ fail eventually because the randomiser itself doesn't take into account that it can generate the same random name twice (or a name chosen from the pre-set list). The only reason this test passes is because 100 is simply insignificant to the total amount of options provided by 8 random letters.

Reply:
Resolving this in favor of the other thread